### PR TITLE
feat(pie-monorepo): DSW-000 added dangerjs for PR checks

### PR DESCRIPTION
**Pull request by ashleynolan**
## Describe your changes

- [Added] - DangerJS added to the repo for PR checks

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code

### CODE REVIEW

File: dangerfile.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
+/* global danger, fail, message */
+import { message, danger } from 'danger';
+
+const modifiedMD = danger.git.modified_files.join('- ');
+message(`Changed Files in this PR: \n - ${modifiedMD}`);
+
+// const { pr } = danger.github;
```
Comment (jamieomaguire) [inline dangerfile.js:7]: Why are we adding all this commented out code?

Comment (ashleynolan) [reply dangerfile.js:7]: Because it's a hack and I haven't added anything useful to it yet or tidied it up 😜

Comment (jamieomaguire) [reply dangerfile.js:7]: ffs, sorry I thought this was a real PR. Ignore me.

File: .github/workflows/dangerjs-checks.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Setup Repo
+        uses: ./.github/actions/setup-repo
+        with:
+          node-version: 18
+          os: ubuntu-latest
+      # Run Danger Checks
+      - name: Danger
+        run: yarn danger ci --failOnErrors
+        env:
+          DANGER_GITHUB_API_TOKEN: ${{ secrets.DANGER_GITHUB_API_TOKEN }}
```
Comment (siggerzz) [inline .github/workflows/dangerjs-checks.yml:26]: I _think_ if you change this to `${{ secrets.CHANGESETS_TOKEN }}`, the comment will be left by the PIE bot :).

We should probably rename that token to something else, but should do the trick!

File: dangerfile.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+/* global danger, fail, message */
+import { message, danger } from 'danger';
+
+const validChangesetCategories = ['Added', 'Changed', 'Removed', 'Fixed'];
+
+// Check for correct Changeset formatting
+danger.git.created_files.filter((filepath) => filepath.includes('.changeset/'))
```
Comment (siggerzz) [inline dangerfile.js:7]: Might also be worth executing this as part of a Git hook just to save people having to do another commit.

We can still have it in this file though on the off-chance someone has disabled husky hooks :)

Comment (ashleynolan) [reply dangerfile.js:7]: Yep – happy to make a ticket for this as an engineering improvement to-do.

It's probably worth testing it, as it does take a little while to run when I've been testing it, so it could get a bit annoying if it makes valid commits hang every time.

File: .github/workflows/dangerjs-checks.yml (hunk 2 above)
Comment (ashleynolan) [reply .github/workflows/dangerjs-checks.yml:26]: It's already hooked up to use the pie bot (as made a new PAT for danger).

If we want to consolidate the tokens if they have the same permissions in the future, could do that too as a bit of a tidy-up?

File: .github/workflows/dangerjs-checks.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+jobs:
+  danger-checks:
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout the Repo
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```
Comment (siggerzz) [inline .github/workflows/dangerjs-checks.yml:15]: `fetch-depth` can be removed. This is only needed for workflows that require full git history of branches / tags etc (like changesets)

**Review by siggerzz — CHANGES_REQUESTED** [review]
A couple minor things, but looks good otherwise!

File: .github/workflows/dangerjs-checks.yml (hunk 4 above)
Comment (ashleynolan) [reply .github/workflows/dangerjs-checks.yml]: Done!

File: .github/workflows/dangerjs-checks.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      # Setup Repo
+      - name: Setup Repo
+        uses: ./.github/actions/setup-repo
+        with:
+          node-version: 18
```
[the remark this answers is not in the document]
Comment (ashleynolan) [reply .github/workflows/dangerjs-checks.yml]: Done!

**Review by kevinrodrigues — APPROVED** [review]

**Review by siggerzz — APPROVED** [review]
